nautobot/core/api/serializers.py

```diff
@@ -152,6 +152,8 @@ def to_representation(self, value):
         url = super().to_representation(value)
         if self.queryset:
             model = self.queryset.model
+        elif isinstance(value, Model):
```

Comment:
This fixes the mishandling of ManyRelatedField from the slack thread https://networktocode-llc.slack.com/archives/C01RMAGBD29/p1686671551128129

Reply:
Fixed more generally in #3916.